netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -667,42 +667,49 @@ net_ebpf_ext_layer_2_classify(
         net_xdp_ctx.base.data_end = packet_buffer + net_buffer->DataLength;
     }
 
-    if (net_ebpf_extension_hook_invoke_program(attached_client, &net_xdp_ctx, &result) == EBPF_SUCCESS) {
-        switch (result) {
-        case XDP_PASS:
-            if (net_xdp_ctx.cloned_nbl != NULL) {
-                // Drop the original NBL.
-                classify_output->actionType = FWP_ACTION_BLOCK;
-                classify_output->rights &= ~FWPS_RIGHT_ACTION_WRITE;
-
-                // Inject the cloned NBL in receive path.
-                status = _net_ebpf_ext_receive_inject_cloned_nbl(net_xdp_ctx.cloned_nbl, incoming_fixed_values);
-                if (NT_SUCCESS(status))
-                    // If cloned packet could be successfully injected, no need to audit for dropping the original.
-                    // So absorb the original packet.
-                    classify_output->flags |= FWPS_CLASSIFY_OUT_FLAG_ABSORB;
-            }
-            // No special processing required in the non-clone case.
-            // The inbound original NBL will be allowed to proceed in the ingress path.
-            break;
-        case XDP_TX:
-            _net_ebpf_ext_handle_xdp_tx(&net_xdp_ctx, incoming_fixed_values);
-            // Absorb the original NBL.
-            classify_output->actionType = FWP_ACTION_BLOCK;
-            classify_output->rights &= ~FWPS_RIGHT_ACTION_WRITE;
-            classify_output->flags |= FWPS_CLASSIFY_OUT_FLAG_ABSORB;
-            break;
-        case XDP_DROP:
+    if (net_ebpf_extension_hook_invoke_program(attached_client, &net_xdp_ctx, &result) != EBPF_SUCCESS) {
+        // Perform a default action if the program fails.
+        result = XDP_DROP;
```

Comment:
Should we add a trace stating that program invocation failed?

Reply:
Hmm, the precedent here isn't entirely clear. It's potentially going to be a noisy trace if the program fails for a long time, but it's also a critical trace if we're wondering why the eBPF program actions aren't taking effect. I'd lean towards not handling this here and hoping that eBPF core has logging that is useful without being noisy, e.g. throttled logs.